packages/nextjs/test/performance/client.test.ts

```diff
@@ -28,28 +30,156 @@ describe('client', () => {
   });
 
   describe('nextRouterInstrumentation', () => {
+    const originalGlobalDocument = getGlobalObject<Window>().document;
+
+    function setUpNextPage(pageProperties: {
+      url: string;
+      route: string;
+      query: any;
+      props: any;
+      hasNextData: boolean;
+    }) {
+      const dom = new JSDOM(
+        // Just some example what a __NEXT_DATA__ tag might look like
+        pageProperties.hasNextData
+          ? `<body>
+          <script id="__NEXT_DATA__" type="application/json">
+          {
+            "props": ${JSON.stringify(pageProperties.props)},
+            "page": "${pageProperties.route}",
+            "query": ${JSON.stringify(pageProperties.query)},
+            "buildId": "y76hvndNJBAithejdVGLW",
+            "isFallback": false,
+            "gssp": true,
+            "appGip": true,
+            "scriptLoader": []
+          }
```

Comment:
nit: Prefer if we create an object and stringify it for readability + we can type with the type exported from nextjs.

Reply:
👍 651e8e0652639c2589cbd4ff57bddb7450dc675c